src/parser/shaman/elemental/modules/azerite/LavaShock.tsx

```diff
@@ -16,25 +19,27 @@ class LavaShock extends Analyzer {
     statTracker: StatTracker,
   };
 
-  procs = 0;
-  damageGained = 0;
-  traitBonus = 0;
+  protected readonly statTracker!: StatTracker;
+  protected procs = 0;
+  protected damageGained = 0;
+  protected traitBonus = 0;
 
-  constructor(...args) {
-    super(...args);
-    this.active = this.selectedCombatant.hasTrait(SPELLS.LAVA_SHOCK.id);
+  constructor(options: any) {
+    super(options);
+    this.active = Boolean(this.selectedCombatant.hasTrait(SPELLS.LAVA_SHOCK.id));
     if (!this.active) {
       return;
     }
     this.traitBonus = this.selectedCombatant.traitsBySpellId[SPELLS.LAVA_SHOCK.id]
-      .reduce((sum, rank) => sum + calculateAzeriteEffects(SPELLS.LAVA_SHOCK.id, rank)[0], 0);
+      .reduce((sum: number, rank: number) => sum + calculateAzeriteEffects(SPELLS.LAVA_SHOCK.id, rank)[0], 0);
   }
 
-  on_byPlayer_damage(event) {
-    if (event.ability.guid !== SPELLS.EARTH_SHOCK.id) {
+  on_byPlayer_damage(event: DamageEvent) {
+    if (event.ability.guid !==  SPELLS.EARTH_SHOCK.id) {
       return;
     }
-    const buff = this.selectedCombatant.getBuff(SPELLS.LAVA_SHOCK_BUFF.id);
+    const buff:any = this.selectedCombatant.getBuff(SPELLS.LAVA_SHOCK_BUFF.id);
```

Comment:
this type can be omitted as it can be implied

Reply:
If I omit the "any", it gets implicitly set to object, which means that it will not allow me to do buff.stacks.

Reply:
Prolly the best idea to just properly specify the getBuff.